### PR TITLE
fix: Memcached persistence

### LINKED ISSUE #14028: Helm: persistent cache item_size_max: 5242880 cannot be larger than ext_wbuf_size: 4194304
**Describe the bug**
In [the latest helm chart](https://github.com/grafana/loki/pull/13619) that adds persistent memcached, the default chart values lead to a failure of memcached to start with:
`persistent cache item_size_max: 5242880 cannot be larger than ext_wbuf_size: 4194304
`
This is on both the chunks and results cache

This will be because the item_size_max is set to 5mb.

Additionally, I can see the template does the following:
`
- --extended=modern,track_sizes{{ if .persistence.enabled }},ext_path={{ .persistence.mountPath }}/file:{{ .persistence.storageSize }}{{ end }}{{ with .extraExtendedOptions }},{{ . }}{{ end }}
`
However the docs on this at https://github.com/memcached/memcached/wiki/ConfiguringLokiExtstore indicate a few more settings need changing or optimising such as:
- ext_wbuf_size (the cause of this issue I think)
- ext_threads
- ext_max_sleep
- slab_automove_freeratio

Simply enabling ext cache without doing any of the other performance tweaks I worry could lead users to a subpar experience? 
This should at least be documented somewhere if the feat isn't 'complete'?

**To Reproduce**
Set chunksCache.persistence.enabled to true

**Expected behavior**
The chunks cache and results memcached starts up with persistence enabled

### PR DESCRIPTION
**Pull request by AndreZiviani**
**What this PR does / why we need it**:
Fixes the usage of persistence on Memcached:
 - Memcached must create a file smaller than disk size
 - `ext_wbuf_size` must be bigger than max item size and be divisible by `wbuf_size` which is 64 by default
 - Add pod permissions to access local volume

**Which issue(s) this PR fixes**:
Fixes #14028
Related #13619 

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [x] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

**Comment by AndreZiviani** [comment]
hey @chaudum please review this when you can

**Comment by pentago** [comment]
why is this taking ages to merge?

**Comment by eric-engberg** [comment]
I don't think ext_wbuf_size should be hard coded.  Users should be able to adjust this if needed

**Comment by AndreZiviani** [comment]
I agree @eric-engberg but I could not think of a way to:
- make it configurable by the user if needed
- automatically add required options for persistency

any suggestions on how to do it?

**Review by chaudum — APPROVED** [review]
@AndreZiviani  Thanks! LGMT
I will merge once the merge conflicts are resolved.

**Comment by chaudum** [comment]
> > I agree @eric-engberg but I could not think of a way to:
> > 
> > * make it configurable by the user if needed
> > * automatically add required options for persistency
> > 
> > any suggestions on how to do it?
> 
> Just add it as a field in the values file. Under persistence maybe?

@eric-engberg Do you want to do that in a separate PR?

**Comment by AndreZiviani** [comment]
since this is a somewhat complex and advanced usage I believe we shouldn't add any options automatically (I had some issued with this on loki-distributed helm), I've added it on previous PR because it looked like it was going to be simple but it is not

**Comment by chaudum** [comment]
@AndreZiviani Seems like the branch needs to be updated again to resolve the conflicts. I'm ready to merge then.

**Comment by AndreZiviani** [comment]
@chaudum done

**Comment by AndreZiviani** [comment]
also looks like the CI for helm release is broken
https://github.com/grafana/loki/issues/14258

**Comment by chaudum** [comment]
Thanks @AndreZiviani !

**Comment by AndreZiviani** [comment]
release still failing btw
https://github.com/grafana/loki/actions/runs/11037391725

**Pull request merged**
